src/LaunchDarkly.Client/EventProcessor.cs

```diff
@@ -82,38 +86,48 @@ void IStoreEvents.Flush()
                 _httpClient?.Dispose();
                 _httpClient = _config.HttpClient();
 
-                Logger.LogDebug("Error sending events: " + Util.ExceptionMessage(e) +
-                                " waiting 1 second before retrying.");
+                Logger.LogDebug(e, 
+                    "Error sending events: {0} waiting 1 second before retrying.",
+                    Util.ExceptionMessage(e));
+
                 Task.Delay(TimeSpan.FromSeconds(1)).Wait();
                 cts = new CancellationTokenSource(_config.HttpClientTimeout);
                 try
                 {
-                    Logger.LogDebug("Submitting " + events.Count + " events to " + _uri.AbsoluteUri + " with json: " +
-                                    jsonEvents);
+                    Logger.LogDebug("Submitting {0} events to {1} with json: {2}",
+                        events.Count,
+                        _uri.AbsoluteUri,
+                        jsonEvents);
                     await SendEventsAsync(jsonEvents, cts);
                 }
                 catch (TaskCanceledException tce)
                 {
                     if (tce.CancellationToken == cts.Token)
                     {
                         //Indicates the task was cancelled by something other than a request timeout
-                        Logger.LogError(string.Format("Error Submitting Events using uri: '{0}' '{1}'", _uri.AbsoluteUri,
-                                            Util.ExceptionMessage(tce)) + tce + " " + tce.StackTrace);
```

Comment:
This is not how you log the StackTrace.  For consumers of this client that want the StackTrace, that's why we pass the full exception details, tce through.  That way we have the full details and stack traces too all the inner exceptions as well.